test/e2e/cmd/run/job.go

```diff
@@ -53,6 +54,18 @@ func NewJob(podName, templatePath string, writer io.Writer, timestampExtractor t
 	}
 }
 
+// Stop is only a best effort to stop the streaming process
+func (j *Job) Stop() {
+	if j.stopRequested {
+		return
+	}
+	j.stopRequested = true
+	close(j.stopLogStream)
+	log.Info("Waiting for log stream to be over", "name", j.jobName)
+	j.logStreamWg.Wait()
+	close(j.streamErrors)
```

Comment:
It doesn't work. We are stuck at `j.logStreamWg.Wait()` because I think we waiting for the next token to scan ([code](https://github.com/elastic/cloud-on-k8s/blob/cf896c7029f70b7682e20115c3b233e9ba346224/test/e2e/cmd/run/run.go#L670-L697)) and never reach `"Log stream ended"`.

Reply:
Can you then just close the log stream instead.

Reply:
What do you have in mind to close the log stream? Is it closing the `io.ReadCloser` like this?

https://github.com/elastic/cloud-on-k8s/blob/cf896c7029f70b7682e20115c3b233e9ba346224/test/e2e/cmd/run/run.go#L666

Reply:
Sorry I meant just closing the log stream channel and ignoring the error channel as you proposed earlier.